desktop/package/package.gradle

```diff
@@ -286,23 +286,89 @@ task packageInstallers {
                         " com/github/sarxos/webcam/ds/buildin/lib/darwin_universal/libOpenIMAJGrabber.dylib"
                 )
 
+                // macOS step 1: Sign dylibs and replace them in the shadow jar
                 // Extract dylibss for signing
                 executeCmd("cd ${tempDylibFolderPath} &&" +
                         " jar xf ${fatJarFolderPath}/${mainJarName}" +
                         dylibsToSign)
-
                 // Sign them
                 executeCmd("cd ${tempDylibFolderPath} &&" +
                         " codesign -vvv --options runtime --deep --force --sign \"${envVariableSigningID}\"" +
                         dylibsToSign)
-
                 // Replace unsigned files in jar file
                 executeCmd("cd ${tempDylibFolderPath} &&" +
                         " jar uf ${fatJarFolderPath}/${mainJarName}" +
                         dylibsToSign)
-            }
 
-            executeCmd(jPackageFilePath + commonOpts + macOpts + " --type dmg")
+                // macOS step 2: Build app-image using the shadow jar above (containing signed dylibs)
+                // NOTE: licensing file cannot be added at this point only when creating the dmg later
+                executeCmd(jPackageFilePath +
+                        " --dest \"${binariesFolderPath}\"" +
+                        " --name Bisq" +
+                        " --description \"A decentralized bitcoin exchange network.\"" +
+                        " --app-version ${appVersion}" +
+                        " --copyright \"© 2020 Bisq\"" +
+                        " --vendor Bisq" +
+                        " --temp \"${jpackateTempDir}\"" +
+                        " --input ${fatJarFolderPath}" +
+                        " --main-jar ${mainJarName}" +
+                        " --main-class bisq.desktop.app.BisqAppMain" +
+                        " --java-options -Xss1280k" +
+                        " --java-options -XX:MaxRAM=4g" +
+                        " --java-options -Djava.net.preferIPv4Stack=true" +
+                        macOpts +
+                        " --type app-image")
+
+                // macOS step 3: Sign app (hardended runtime)
+                File bisqAppImageFullPath = new File(binariesFolderPath, "Bisq.app")
+                executeCmd("codesign" +
+                        " --sign \"${envVariableSigningID}\"" +
+                        " --options runtime" +
+                        " --entitlements macos.entitlements" +
+                        " --force" +
+                        " --verbose" +
+                        "${bisqAppImageFullPath}/Contents/runtime/Contents/MacOS/libjli.dylib")
+                executeCmd("codesign" +
+                        " --sign \"${envVariableSigningID}\"" +
+                        " --options runtime" +
+                        " --entitlements macos.entitlements" +
```

Comment:
I think you missed to add the `macos.entitlements` file to the build.

Reply:
Done